src/pokemon_storage_system_2.c

```diff
@@ -0,0 +1,653 @@
+    }
+    for (boxId = 0; boxId < TOTAL_BOXES_COUNT; boxId++)
+    {
+        SetBoxWallpaper(boxId, boxId % 4);
+    }
+}
+
+void LoadBoxSelectionPopupSpriteGfx(struct UnkPSSStruct_2002370 *a0, u16 tileTag, u16 palTag, u8 a3, bool32 loadPal)
+{
+    struct SpritePalette palette = {
+        sBoxSelectionPopupPalette, palTag
+    };
+    struct SpriteSheet sheets[] = {
+        {sBoxSelectionPopupCenterTiles, 0x800, tileTag},
+        {sBoxSelectionPopupSidesTiles,  0x180, tileTag + 1},
+        {}
+    };
+
+    if (loadPal)
+        LoadSpritePalette(&palette);
+
+    LoadSpriteSheets(sheets);
+    sBoxSelectionPopupSpriteManager = a0;
+    a0->tilesTag = tileTag;
+    a0->paletteTag = palTag;
+    a0->subpriority = a3;
+    a0->loadPal = loadPal;
+}
+
+void FreeBoxSelectionPopupSpriteGfx(void)
+{
+    if (sBoxSelectionPopupSpriteManager->loadPal)
+        FreeSpritePaletteByTag(sBoxSelectionPopupSpriteManager->paletteTag);
+    FreeSpriteTilesByTag(sBoxSelectionPopupSpriteManager->tilesTag);
+    FreeSpriteTilesByTag(sBoxSelectionPopupSpriteManager->tilesTag + 1);
+}
+
+void sub_808C940(u8 curBox)
+{
+    sub_808C9C4(curBox);
+}
+
+void sub_808C950(void)
+{
+    sub_808CBA4();
+}
+
+u8 HandleBoxChooseSelectionInput(void)
+{
+    if (JOY_NEW(B_BUTTON))
+    {
+        PlaySE(SE_SELECT);
+        return 201;
+    }
+    if (JOY_NEW(A_BUTTON))
+    {
+        PlaySE(SE_SELECT);
+        return sBoxSelectionPopupSpriteManager->curBox;
+    }
+    if (JOY_NEW(DPAD_LEFT))
+    {
+        PlaySE(SE_SELECT);
+        UpdateBoxNameAndCountSprite_WraparoundLeft();
+    }
+    else if (JOY_NEW(DPAD_RIGHT))
+    {
+        PlaySE(SE_SELECT);
+        UpdateBoxNameAndCountSprite_WraparoundRight();
+    }
+    return 200;
+}
+
+static const union AnimCmd gUnknown_83CDA50[] = {
+    ANIMCMD_FRAME( 0, 5),
+    ANIMCMD_END
+};
+
+static const union AnimCmd gUnknown_83CDA58[] = {
+    ANIMCMD_FRAME( 4, 5),
+    ANIMCMD_END
+};
+
+static const union AnimCmd gUnknown_83CDA60[] = {
+    ANIMCMD_FRAME( 6, 5),
+    ANIMCMD_END
+};
+
+static const union AnimCmd gUnknown_83CDA68[] = {
+    ANIMCMD_FRAME(10, 5),
+    ANIMCMD_END
+};
+
+static const union AnimCmd *const gUnknown_83CDA70[] = {
+    gUnknown_83CDA50,
+    gUnknown_83CDA58,
+    gUnknown_83CDA60,
+    gUnknown_83CDA68
+};
+
+static const union AffineAnimCmd gUnknown_83CDA80[] = {
+    AFFINEANIMCMD_FRAME(224, 224, 0, 0),
+    AFFINEANIMCMD_END
+};
+
+static const union AffineAnimCmd *const gUnknown_83CDA90[] = {
+    gUnknown_83CDA80
+};
+
+static void sub_808C9C4(u8 curBox)
+{
+    u16 i;
+    u8 spriteId;
+    struct SpriteTemplate template;
+    struct OamData oamData = {};
+    oamData.size = SPRITE_SIZE(64x64);
+    oamData.paletteNum = 1;
+    template = (struct SpriteTemplate){
+        0, 0, &oamData, gDummySpriteAnimTable, NULL, gDummySpriteAffineAnimTable, SpriteCallbackDummy
+    };
+    {
+    const u8 gUnknown_83CDA94[] = _("/30");
+
```

Comment:
(1 line above) is there any problem on this comment?

Reply:
Yes: making this static will remove the automatic memcpy.